cirq-google/cirq_google/devices/grid_device.py

```diff
@@ -74,6 +68,80 @@
 _VARIADIC_GATE_FAMILIES = [_MEASUREMENT_GATE_FAMILY, _WAIT_GATE_FAMILY]
 
 
+GateOrFamily = Union[Type[cirq.Gate], cirq.Gate, cirq.GateFamily]
+
+
+@dataclass
+class _GateRepresentations:
+    """Contains equivalent representations of a gate in both DeviceSpecification and GridDevice."""
+
+    """The name of gate type in `GateSpecification`."""
+    gate_spec_name: str
+    """Gate representations to be included in generated gatesets and gate durations."""
+    primary_forms: List[GateOrFamily]
```

Comment:
The primary and additional forms definition is kind of confusing to me.  This new terminology is not explained very clearly by these docstrings.  Maybe the class docstring could give more context as to what's going on here?